examples/std/src/bin/net.rs

```diff
@@ -26,6 +28,41 @@ async fn net_task(mut runner: embassy_net::Runner<'static, TunTapDevice>) -> ! {
     runner.run().await
 }
 
+struct WriteBuf<const N: usize> {
+    len: usize,
+    bytes: [u8; N],
+}
+
+/// In a `no_std` situation it is useful to be able to
+/// format into a fixed-size buffer using the
+/// `core::fmt::write` macro. This struct enables that.
```

Comment:
this is a bit too complex for an example, i'd suggest using `heapless::String` instead. you can `write!` into it just like `WriteBuf`.

Reply:
Doh. Hadn't noticed we were already depending on `heapless` and was trying to avoid it. Thanks for the catch. Fixed.